File: backend/modules/sabor-familia/src/main/java/br/com/puc/saborfamilia/service/restricao/RestricaoAlimentarService.java
package br.com.puc.saborfamilia.service.restricao;

import br.com.puc.saborfamilia.database.entity.RestricaoAlimentarEntity;
import br.com.puc.saborfamilia.service.restricao.dto.RestricaoAlimentarResponse;
import br.com.puc.saborfamilia.service.restricao.dto.request.EditarRestricaoAlimentarRequest;
import java.util.List;

public interface RestricaoAlimentarService {

  RestricaoAlimentarResponse buscarRestricaoAlimentar(String codigo);

  List<RestricaoAlimentarResponse> buscarRestricoesAlimentares();

  RestricaoAlimentarResponse editarRestricaoAlimentar(String codigo, EditarRestricaoAlimentarRequest request);

  RestricaoAlimentarResponse ativarRestricaoAlimentar(String codigo);

  RestricaoAlimentarResponse inativarRestricaoAlimentar(String codigo);

  List<RestricaoAlimentarEntity> validarRestricoesAlimentares(List<String> restricaoCodigos);

}

File: frontend/src/features/sabor_familia/service/RestricaoalimentarService.ts
import api from "../lib/Api";
import type { RestricaoAlimentarResponse } from "../dto/restricao/response/RestricaoAlimentarResponse";

export const restricaoAlimentarService = {
  buscarRestricoesAlimentares: async (): Promise<RestricaoAlimentarResponse[]> => {
    const { data } = await api.get<RestricaoAlimentarResponse[]>("/restricao-alimentar");
    return data;
  },
};
